crypto/signature_cgo.go

```diff
@@ -58,6 +60,22 @@ func Sign(hash []byte, prv *ecdsa.PrivateKey) (sig []byte, err error) {
 	return secp256k1.Sign(hash, seckey)
 }
 
+// VerifySignature checks that the given public key created signature over hash.
+// The public key should be in compressed (33 bytes) or uncompressed (64 bytes) format.
+// The signature should have the 64 byte [R || S] format.
+func VerifySignature(pubkey, hash, signature []byte) bool {
+	return secp256k1.VerifySignature(pubkey, hash, signature)
+}
+
+// DecompressPubkey parses a public key in the 33-byte compressed format.
+func DecompressPubkey(pubkey []byte) (*ecdsa.PublicKey, error) {
+	x, y := secp256k1.DecompressPubkey(pubkey)
+	if x == nil {
```

Comment:
Design question: Why does the underlying method return `nil`, and we create an `error` here, instead of returning error from the lower layer?

Reply:
Compressed/uncompressed doesn't matter for SigToPub because the return value isn't encoded.

Reply:
I decided this way to simplify the interface. There is only one possible error anyway: "invalid". `elliptic.Unmarshal` uses the same convention.

Reply:
You are right though, maybe this should be changed.